### ISSUE TITLE
travis builds are breaking (again)

### ISSUE THREAD

**Issue by Spongman** [processing/p5.js#3402]
travis broke something behind the scenes and builds have been breaking since 12/5:
- https://travis-ci.org/processing/p5.js/builds/464015031

this is quite unfortunate given the sudden increase in PRs recently. every single one is failing for the same reason:
- https://travis-ci.org/processing/p5.js/pull_requests

**Comment by Spongman** [comment]
i just set up a CI azure pipeline for my repo, and it's working fine. so it must be something specific to travis' environment. @BanzaiMan

**Comment by BanzaiMan** [comment]
How is "blob" getting in the URL? https://travis-ci.org/processing/p5.js/jobs/464015032#L596 It is not something Travis CI should be doing.

**Comment by Spongman** [comment]
not sure. don't know that that is actually a problem.

looking at the commit history: https://github.com/processing/p5.js/commits/master i see that the build for d6b811d5e9f5b88d09e2c25db5409676984625eb succeeded, and the build for the *next* commit (777ad7049ed8ef4931a9078871a20fdc51e97ae4) failed.

here's the diff for that commit that 'caused' the failure: https://github.com/processing/p5.js/commit/777ad7049ed8ef4931a9078871a20fdc51e97ae4

you probably don't need me to tell you that isn't the cause of the failure.

i'm at a loss here to find the cause of this failure. again: this build works on every machine i have tried it on, including windows, linux, azure pipeline. i'm currently setting up appveyor to test it on there. i assume it'll work fine.

**Comment by BanzaiMan** [comment]
Try restarting previously passing builds. If they fail now, I suspect changes in underlying dependencies are causing problems. (For example, Chrome was updated from [70.0.3538.110-1](https://travis-ci.org/processing/p5.js/jobs/462464275#L423) to [71.0.3578.80-1](https://travis-ci.org/processing/p5.js/jobs/464015601#L423).)

**Comment by outofambit** [comment]
is it worth looking into switching to circleci from travis? i could put up a PR for that in the next day or two.

**Comment by BanzaiMan** [comment]
How does CircleCI provide older versions of Chrome? I do not think Google distributes anything but the stable one.

**Comment by Spongman** [comment]
yeah, i don't think pinning the version is a good idea, regardless of whether or not it's possible.

**Comment by limzykenneth** [comment]
> can someone else check out the mocha-chrome project (https://github.com/shellscape/mocha-chrome) and, with chrome >= 71 installed, run npm i && npm run bdd.
> 
> actually, if you could run that before upgrading chrome, that would be great, too...

All tests pass with Chrome 71 on macOS Mojave

```
> mocha-chrome@1.1.0 bdd /Users/kennethlim/Dropbox/Sites/mocha-chrome
> mocha test/test.js --timeout 10000 --exit



  MochaChrome
Mocha-Chrome Failed: mocha was not found in the page within 1000ms of the page loading.
Mocha-Chrome Failed: mocha.run() was not called within 1000ms of the page loading.
    ✓ fails if mocha isn't loaded (2037ms)
Mocha-Chrome Failed: mocha.run() was not called within 1000ms of the page loading.
    ✓ fails if mocha.run isn't called (2141ms)
    ✓ runs a test (1413ms)
    ✓ reports a failure (1220ms)
    ✓ allows runner modification (1371ms)
    ✓ supports different reporters (1309ms)
    ✓ supports mixed tests (1402ms)
    ✓ reports async failures (1382ms)
    ✓ supports test using and clearing localStorage (1334ms)
    ✓ handles circular structures in console.log (1268ms)

  mocha-chrome binary
    ✓ should report version (333ms)
    ✓ should run a successful test (1575ms)
    ✓ should run a failing test (1707ms)
    ✓ should default to "spec" reporter (1767ms)
    ✓ should honor --spec parameter (1694ms)
    ✓ should allow use of --chrome-flags (1645ms)
    ✓ should use the --timeout flag value (3055ms)


  17 passing (27s)
```

**Comment by Spongman** [comment]
i have found that the tests start passing again if the following line is commented out:

https://github.com/processing/p5.js/blob/9a0a510dd4f16406d0386459322010ad95874c4b/lib/addons/p5.sound.js#L4514

i think this might be a good fix for now until we find some compatible way to compute the `Tone.Context.lag` property.

**Comment by Spongman** [comment]
i have a hacked-up mocha-chrome.js that uses puppeteer instead. it runs the tests, but it still needs some work to handle edge cases and failures.

https://github.com/Spongman/p5.js/tree/mocha-puppeteer

**Comment by Spongman** [comment]
can someone test that my fix #3415 works on a mac (with and without test failures) ? i have tested on windows, ubuntu and travis/docker.
